api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirDiagnosticReportDaoImpl.java

```diff
@@ -11,19 +11,28 @@
 
 import static org.hibernate.criterion.Restrictions.eq;
 
+import java.util.Collection;
+
 import lombok.AccessLevel;
 import lombok.Setter;
+import org.hibernate.Criteria;
 import org.hibernate.SessionFactory;
 import org.openmrs.Obs;
 import org.openmrs.api.db.DAOException;
 import org.openmrs.module.fhir2.api.dao.FhirDiagnosticReportDao;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirDiagnosticReportDaoImpl implements FhirDiagnosticReportDao {
+public class FhirDiagnosticReportDaoImpl extends BaseFhirDao<Obs> implements FhirDiagnosticReportDao {
+	
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
 	
 	@Autowired
 	@Qualifier("sessionFactory")
```

Comment:
So as part of this PR, you should get rid of this `sessionFactory` instance. That's why the build is failing.

Reply:
Alright , noted that , i think i will close  it , since some one else was simultaneously working on the Ticket at the same time.